### ISSUE TITLE
Upsert : invalid syntaxe with PostgreSQL

### ISSUE THREAD

**Issue by sgallou** [Yadoms/yadoms#741]
"INSERT OR REPLACE INTO" is used to upsert some records. But this syntax is SQLite specific.

Need to specialize this DB command for SQLite and PostgreSQL

**Comment by jano42** [comment]
This description is not a bug !
I don't remember but it is either already specialized by SQL engine, or the syntax is compatible.

If a real bug, please add "error" logs....

**Comment by sgallou** [comment]
See https://www.yadoms.com/forum/viewtopic.php?f=2&t=2480&p=2775#p2775

**Comment by sgallou** [comment]
In CQuery class, a member m_insertOrUpdateName is defined and initialized with "INSERT OR REPLACE INTO".
It's not specific to the SQL engine, but not support by PostgreSQL.

**Comment by jano42** [comment]
It's very strange. I'm sure I tested it heavily.
In my memory the command is sqlite specific but also supported by PostgreSQL. (Maybe this is a feature that is not supported by the Postgresql version)

**Comment by jano42** [comment]
Since SQLite 3.24 now  "support for PostgreSQL-style UPSERT" we can generalize it.
This is a big job to do and test (many impacts ; and heavy "summary" queries use it) 😭😱

**Comment by jano42** [comment]
I've looked at the code, and there is all needed functions do manage InsertOrUpdate


Each database provider (sqlite or pgsql) must implement ``supportInsertOrUpdateStatement```. Currently, SQLite say YES, and PostgreSQL say NO.

Finally the implementation is:
````
if  supportInsertOrUpdateStatement
then call InsertOrUpdate
else call Update, and if it fails, then call InsertInto
endif
````

This is already implemented in https://github.com/Yadoms/yadoms/blob/master/sources/server/database/common/requesters/Acquisition.cpp#L466

**Comment by sgallou** [comment]
Fixed by 3a480cdcffd4affcba7b34432d4e6ac470049c83
